HW_09/weatherApp.js

```diff
@@ -0,0 +1,50 @@
+import { input, list } from "./constants.js";
+
+export class WeatherApp {
+    constructor(API_KEY) {
+        this.API_KEY = API_KEY;
+        this.isLoading = false;
+
+    }
+
+    // fetchData = () => {
+    //     const inputVal = input.value;
+    //     const url = `https://api.openweathermap.org/data/2.5/weather?q=${inputVal}&appid=${this.API_KEY}&units=metric`;
+
+    //     this.isLoading = true;
+    //     fetch(url)
+    //         .then((response) => response.json())
+    //         .then((data) => {
+    //             console.log(data);
+    //             this.updateDOM(data);
+    //         })
+    //         .catch((error) => {/*  */
+    //             throw new Error(error);
+    //         })
+    //         .finally(() => {
+    //             this.isLoading = false;
+    //         })
+    // }
+
+    async fetchData() {
+            const inputVal = input.value;
+            const url = `https://api.openweathermap.org/data/2.5/weather?q=${inputVal}&appid=${this.API_KEY}&units=metric`;
+            const response = await fetch(url);
+            const data = await response.json();
+            console.log(data);
+            this.updateDOM(data);
```

Comment:
fetch завжди має бути обгорнутий в блок try...catch
```suggestion
try {
            const response = await fetch(url);
            const data = await response.json();
            console.log(data);
            this.updateDOM(data);
            } catch(error) {
            throw new Error(error);
            
```

Reply:
Виправила згідно з твоїми рекомендаціями.